### PR TITLE
20913 Gofer-Tests package cleanup (solve critiques)

### PR DESCRIPTION
**Pull request by astares**
- use assert:equals: instead of =
- use cascading where possible
cosmetic changes only, no change of functionality

### CODE REVIEW

File: src/Gofer-Tests/GoferApiTest.class.st [hunk 1]
```diff
@@ -141,13 +154,15 @@ GoferApiTest >> testPackageReference [
...
-	self assert: (gofer resolved first packageName = 'GoferFoo').
-	self assert: (gofer resolved first author = 'lr').
-	self assert: (gofer resolved first branch isEmpty).
-	self assert: (gofer resolved first versionNumber = 4).
-	self assert: (gofer resolved first repository = self monticelloRepository)
+
+	self 
+		assert: gofer resolved size equals: 1;
+		assert: (gofer resolved first isKindOf: GoferResolvedReference);
+		assert: gofer resolved first packageName equals: 'GoferFoo';
```
Comment (jecisc) [inline src/Gofer-Tests/GoferApiTest.class.st]: Could be cool to do the same as in testConstraintReference by saving the reference.

File: src/Gofer-Tests/GoferReferenceTest.class.st [hunk 2]
```diff
@@ -162,46 +180,53 @@ GoferReferenceTest >> testVersionShouldFindWorkingCopy [
...
 GoferReferenceTest >> testVersionShouldParseComplexName [
 	| queryReference |
+	
 	queryReference := GoferVersionReference name: 'Seaside2.8b5'.
-	self assert: queryReference packageName = 'Seaside2'.
-	self assert: queryReference author isEmpty.
-	self assert: queryReference branch = '8b5'.
-	self assert: queryReference versionNumber = 0.
+	self 
+		assert: queryReference packageName equals: 'Seaside2';
```
Comment (pavel-krivanek) [inline src/Gofer-Tests/GoferReferenceTest.class.st]: I (with Denis) think that we should not use cascade on assertions. It makes debugging harder on failing assertions.

Comment (astares) [reply src/Gofer-Tests/GoferReferenceTest.class.st]: Why does it make debugging harder? And if so - shouldnt we then change Debugging?

**Comment by pavel-krivanek** [comment]
It is not direct problem of debugger as a tool. Let's say you have code like this:
```
	self assert: 1 equals: 1;
		assert: 2 equals: 2;
		assert: 2 equals: 1;
		assert: 3 equals: 3.
```
It will fail and the debugger window will be opened on the window. The issue we talk about is that it the debugger you cannot select and evaluate the next (or previous) assertion because the line with it is part of a bigger cascade. 
It is just cosmetic issue but can be annoying.

**Comment by astares** [comment]
The debugger will stop right at the failing assertion and you can just step into them
as this picture proves:

image

If one needs to reevaluate you can select the whole cascase and reevaluate by running
(to stop at the failing) or step through as usual.

A cascade is more compact and more readable as you quickly see that the messages
go to the same object. Which helps to better and easier understand the code (here
a test).

**Comment by astares** [comment]
Have a look at the "horrible" RPackage tests for instance. Same issue: they are VERY hard to understand the way the code was written ... a few renamings and cascades grouping the assertions already makes it much more clear what each method does.

**Comment by Ducasse** [comment]
Hi torsten

Pay attention with the cascade in the tests. Denis was working with me on a bug in Sunit tests and the first things he did was to remove the cascade so that he could work. I see that pavel is saying the same. So we should pay attention. 

Stef

**Pull request merged**